### PR TITLE
Assorted bug fixes for 0.8

### PR DESCRIPTION
**Pull request by rshriram**
Fix incorrect mixer filter generation.
Fix incorrect comparison/FQDN expansion of mesh default gateway ("mesh").
Do not generate listeners for non TCP protocols

Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

### CODE REVIEW

File: pilot/pkg/networking/plugin/mixer/mixer.go [hunk 1]
```diff
@@ -56,13 +56,17 @@ func (Plugin) OnOutboundListener(in *plugin.InputParams, mutable *plugin.Mutable
...
+			if m != nil {
+				mutable.FilterChains[cnum].HTTP = append(mutable.FilterChains[cnum].HTTP, m)
+			}
 		}
 		return nil
 	case plugin.ListenerTypeTCP:
-		for cnum := range mutable.FilterChains {
-			mutable.FilterChains[cnum].TCP = append(mutable.FilterChains[cnum].TCP, buildMixerOutboundTCPFilter(env, node))
-		}
+		// Adding an empty filter prevents listeners from loading
```
Comment (ostromart) [inline pilot/pkg/networking/plugin/mixer/mixer.go]: just do same nil check as above?

File: pilot/pkg/networking/core/v1alpha3/listener.go [hunk 2]
```diff
@@ -355,23 +364,23 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListeners(env model.En
...
-					log.Warnf("Multiple TCP listener definitions for %s", listenerMapKey)
+				if n, exists := listenerMap[listenerMapKey]; exists {
+					log.Warnf("Multiple TCP listener definitions for %s %v %s", listenerMapKey, n, service.Hostname)
 					continue
 				}
 				listenerOpts.filterChainOpts = []*filterChainOpts{{
 					networkFilters: buildOutboundNetworkFilters(clusterName, addresses, servicePort),
 				}}
+			default:
+				log.Infof("buildSidecarOutboundListeners: service %q has unknown protocol %#v", service.Hostname, servicePort)
```
Comment (ostromart) [inline pilot/pkg/networking/core/v1alpha3/listener.go]: warn

**Comment by hklai** [comment]
Are you going to cherrypick this to the release branch? https://github.com/istio/istio/commits/release-0.8

**Pull request merged**